File: Java/Sandbox/test/com/codesignal/ArrayMaximalAdjacentDifferenceTests.java
package com.codesignal;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class ArrayMaximalAdjacentDifferenceTests {
    @Test
    public void test1(){
        Assertions.assertEquals(30, Kata.arrayMaximalAdjacentDifference(new int[]{-14, 15, -15}));
    }

    @Test
    public void test2(){
        Assertions.assertEquals(4, Kata.arrayMaximalAdjacentDifference(new int[]{-1, 1, -3, -4}));
    }

    @Test
    public void test3(){
        Assertions.assertEquals(0, Kata.arrayMaximalAdjacentDifference(new int[]{-2, -2, -2, -2, -2}));
    }
}

File: SolutionForFun/test/CodeWarsTest/Tests/GeneralKataTests/ArrayMaximalAdjacentDifferenceTests.cs
using CodeWarsTests.Tasks;
using NUnit.Framework;

namespace CodeWarsTests.Tests.GeneralKataTests
{
    [TestFixture]
    public class ArrayMaximalAdjacentDifferenceTests
    {
        [Test]
        public void Test1()
        {
            Assert.AreEqual(3, Kata.ArrayMaximalAdjacentDifference(new[] { 2, 4, 1, 0 }));
        }

        [Test]
        public void Test2()
        {
            Assert.AreEqual(0, Kata.ArrayMaximalAdjacentDifference(new[] { 1, 1, 1, 1 }));
        }

        [Test]
        public void Test3()
        {
            Assert.AreEqual(7, Kata.ArrayMaximalAdjacentDifference(new[] { -1, 4, 10, 3, -2 }));
        }

        [Test]
        public void Test4()
        {
            Assert.AreEqual(2, Kata.ArrayMaximalAdjacentDifference(new[] { 10, 11, 13 }));
        }
    }
}
